core-backend/sandbox/src/env.rs

```diff
@@ -71,6 +71,33 @@ pub(crate) struct Runtime<E: Ext> {
     pub termination_reason: Option<TerminationReasonKind>,
 }
 
+// A helping wrapper for `EnvironmentDefinitionBuilder` and `forbidden_funcs`.
+// It makes adding functions to `EnvironmentDefinitionBuilder` shorter.
+struct EnvBuilder<'a, E: Ext> {
+    env_def_builder: EnvironmentDefinitionBuilder<Runtime<E>>,
+    forbidden_funcs: &'a BTreeSet<&'static str>,
+}
+
+impl<'a, E: Ext + IntoExtInfo + 'static> EnvBuilder<'a, E> {
+    fn add_func(&mut self, name: &str, f: HostFuncType<Runtime<E>>)
+    where
+        E::Error: AsTerminationReason + IntoExtError,
+    {
+        if self.forbidden_funcs.contains(name) {
+            self.env_def_builder
+                .add_host_func("env", name, crate::funcs::FuncsHandler::forbidden);
```

Comment:
```suggestion
                .add_host_func("env", name, FuncsHandler::forbidden);
```

Move import on top of the file

Reply:
Yes, my fault, fixed.